### PR TITLE
Stairs DLC: Buildable stairs, material datum staircases, fall up a staircase

### PR DESCRIPTION
**Pull request by Paxilmaniac**
## About The Pull Request
Adds a few new types of stairs, and makes stairs buildable within rounds.
Also removes the terminator sprite variation for stairs, because its basically unused and really not needed with plane cube multiz.

A fine example of what this looks like:

https://user-images.githubusercontent.com/82386923/195580735-3759c833-6283-455b-9d82-4695ae27991f.mp4

## Why It's Good For The Game
Currently ladders are the only way to build between z-levels, the ability to build stairs is a step in the right direction for making interacting with multiz in a round more accessible to players. Material datum stairs are also simply funny, who doesn't want a grand staircase made out of gold/plasma/uranium/pizza/whatever else?

## Changelog

:cl:
add: Stairs can now be built out of iron, wood, or really any rigid material (stuff you can make carving blocks out of)
add: Two new types of stairs, wood and stone (wood is buildable in round) for any mappers in the chat
/:cl:

### CODE REVIEW

**Review by Fikou — CHANGES_REQUESTED** [review]
make this a more involved process with a frame made out of rods or something that you have to fill in with a material

**Comment by Paxilmaniac** [comment]
I could do that maybe

**Comment by Mickyan** [comment]
Unlike ladders which require previous access to both z-levels to build, these can be used to climb over any wall using something you can hold in your backpack and is quite literally everywhere

Currently this will break the intended flow of every multi-z map (like climbing out of perma on tram) and make verticality a trivial obstacle

**Comment by Jacquerel** [comment]
Two requests:
- Can this require tools so that prisoners can't just build stairs to climb out of prison (even if it's just into the security lobby that is a pretty easy way to be a big pain in the ass)
- Can this have a "snow stairs" recipe for ice box which only requires snow (and tools)? Or is this already covered under "material stairs"?

**Comment by Paxilmaniac** [comment]
I'm gonna make it require at least a wrench to build

**Comment by Paxilmaniac** [comment]
As per request, building stairs is now a process of build a frame > wrenching it > applying material

https://user-images.githubusercontent.com/82386923/195601744-7f36eff7-707f-4992-a834-cb6610640b2b.mp4

**Comment by optimumtact** [comment]
They check the floor above has a hole where the stairs will go right?

**Comment by Paxilmaniac** [comment]
Yes they do

File: code/game/objects/structures/stairs.dm [hunk 1]
```diff
@@ -150,3 +156,100 @@
...
+			var/list/material_list = list()
+			if(material.material_type)
+				material_list[material.material_type] = MINERAL_MATERIAL_AMOUNT * 10
+			make_new_stairs(/obj/structure/stairs/material, material_list)
+		return TRUE
+	return ..()
+
+/obj/structure/stairs_frame/proc/make_new_stairs(stairs_type, custom_materials)
+	var/obj/structure/stairs/new_stairs = new stairs_type(loc)
+	new_stairs.dir = dir
```
Comment (Fikou) [inline code/game/objects/structures/stairs.dm:252]: use setDir

File: code/game/objects/structures/stairs.dm [hunk 2]
```diff
@@ -150,3 +156,100 @@
...
+	new frame_stack(get_turf(src), frame_stack_amount)
+	qdel(src)
+
+/obj/structure/stairs_frame/attackby(obj/item/attacked_by, mob/user, params)
+	if(!anchored)
+		if(isstack(attacked_by))
+			user.balloon_alert(user, "secure frame first")
+			return TRUE
+		return ..()
+	if(isstack(attacked_by))
```
Comment (Fikou) [inline code/game/objects/structures/stairs.dm:220]: early return here

**Review by Fikou — APPROVED** [review]
testmerged to see how it plays out

**Comment by Colovorat** [comment]
Can we get much higher higher... So high...

**Pull request merged**